BUILD.gn

```diff
@@ -0,0 +1,507 @@
+      "//chrome/renderer/pepper/pepper_flash_font_file_host.h",
+      "//chrome/renderer/pepper/pepper_flash_fullscreen_host.cc",
+      "//chrome/renderer/pepper/pepper_flash_fullscreen_host.h",
+      "//chrome/renderer/pepper/pepper_flash_menu_host.cc",
+      "//chrome/renderer/pepper/pepper_flash_menu_host.h",
+      "//chrome/renderer/pepper/pepper_flash_renderer_host.cc",
+      "//chrome/renderer/pepper/pepper_flash_renderer_host.h",
+      "runtime/browser/renderer_host/pepper/xwalk_browser_pepper_host_factory.cc",
+      "runtime/browser/renderer_host/pepper/xwalk_browser_pepper_host_factory.h",
+      "runtime/renderer/pepper/pepper_helper.cc",
+      "runtime/renderer/pepper/pepper_helper.h",
+      "runtime/renderer/pepper/pepper_uma_host.cc",
+      "runtime/renderer/pepper/pepper_uma_host.h",
+      "runtime/renderer/pepper/xwalk_renderer_pepper_host_factory.cc",
+      "runtime/renderer/pepper/xwalk_renderer_pepper_host_factory.h",
+    ]
+    deps += [
+      "//components/pdf/renderer",
+      "//ppapi/host",
+      "//ppapi/proxy",
+      "//ppapi/proxy:ipc",
+      "//ppapi/shared_impl",
+    ]
+  }
+
+  if (toolkit_views) {
+    sources += [
+      "runtime/browser/ui/xwalk_javascript_native_dialog_factory.h",
+      "runtime/browser/ui/xwalk_views_delegate.cc",
+      "runtime/browser/ui/xwalk_views_delegate.h",
+    ]
+    deps += [
+      "//components/app_modal",
+      "//components/constrained_window",
+      "//components/guest_view/browser",
+      "//components/ui/zoom",
+      "//ui/events",
+      "//ui/resources",
+      "//ui/strings:ui_strings",
+      "//ui/views",
+      "//ui/views/controls/webview",
+    ]
+  }
+
+  if (use_aura) {
+    deps += [
+      "//ui/aura",
+      "//ui/base/ime",
+      "//ui/wm",
+    ]
+  }
+
+  if (use_webui_file_picker) {
+    defines += [ "USE_WEBUI_FILE_PICKER" ]
+    sources += [
+      "runtime/browser/ui/browser_dialogs.h",
+      "runtime/browser/ui/linux_webui/linux_webui.cc",
+      "runtime/browser/ui/linux_webui/linux_webui.h",
+      "runtime/browser/ui/linux_webui/select_file_dialog_impl_webui.cc",
+      "runtime/browser/ui/linux_webui/select_file_dialog_impl_webui.h",
+      "runtime/browser/ui/webui/file_picker/file_picker_ui.cc",
+      "runtime/browser/ui/webui/file_picker/file_picker_ui.h",
+      "runtime/browser/ui/webui/file_picker/file_picker_web_dialog.cc",
+      "runtime/browser/ui/webui/file_picker/file_picker_web_dialog.h",
+      "runtime/browser/ui/webui/xwalk_web_contents_handler.cc",
+      "runtime/browser/ui/webui/xwalk_web_contents_handler.h",
+      "runtime/browser/ui/webui/xwalk_web_ui_controller_factory.cc",
+      "runtime/browser/ui/webui/xwalk_web_ui_controller_factory.h",
+      "runtime/browser/ui/xwalk_web_dialog_view.cc",
+      "runtime/common/url_constants.cc",
+      "runtime/common/url_constants.h",
+    ]
+    deps += [ "//content/app/resources" ]
+  }
+}
+
+action("generate_upstream_blink_version") {
```

Comment:
The comment above this action's gyp version ("While we could just call lastchange.py [...]") is pretty important, so it should be added here too.